pkg/readers/container_image_reader.go

```diff
@@ -0,0 +1,148 @@
+package readers
+
+import (
+	"context"
+	"fmt"
+	scalibr "github.com/google/osv-scalibr"
+	scalibrlayerimage "github.com/google/osv-scalibr/artifact/image/layerscanning/image"
+	"github.com/google/osv-scalibr/converter"
+	"github.com/google/osv-scalibr/extractor"
+
+	//"github.com/google/osv-scalibr/converter"
+	el "github.com/google/osv-scalibr/extractor/filesystem/list"
+	sl "github.com/google/osv-scalibr/extractor/standalone/list"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/parser"
+)
+
+type ContainerImageReaderConfig struct {
+	Image string
+}
+
+type containerImageReader struct {
+	config *ContainerImageReaderConfig
+}
+
+var _ PackageManifestReader = &containerImageReader{}
+
+func NewContainerImageReader(config *ContainerImageReaderConfig) (*containerImageReader, error) {
+	return &containerImageReader{
+		config: config,
+	}, nil
+}
+
+func (c containerImageReader) Name() string {
+	return "Container Image Reader"
+}
+
+func (c containerImageReader) ApplicationName() (string, error) {
+	return defaultApplicationName, nil
+}
+
+func (c containerImageReader) EnumManifests(handler func(*models.PackageManifest, PackageReader) error) error {
+	image, err := c.getScalibrContainerImage()
+	if err != nil {
+		logger.Errorf("failed to get Scalibr container image: %s", err)
+		return fmt.Errorf("failed to get Scalibr container image: %s", err)
+	}
+
+	scanConfig, err := c.getScalibrScanConfig()
+	if err != nil {
+		logger.Errorf("failed to get scan config: %s", err)
+		return fmt.Errorf("failed to get Scalibr scan config: %s", err)
+	}
+
+	// Scan Container
+	result, err := scalibr.New().ScanContainer(context.Background(), image, scanConfig)
+	if err != nil {
+		logger.Errorf("failed to perform osv-scalibr scan: %s", err)
+		return fmt.Errorf("failed to perform osv-scalibr scan: %s", err)
+	}
+
+	// Handle duplicated packages as same packages (name+version) are found on different locations.
+	// For golang:alpine, 25 % of packages were repeated: the total was 474 with a unique 355
+	packageSet := make(map[string]*extractor.Package)
+
+	manifests := make(map[string]*models.PackageManifest)
+
+	for _, pkg := range result.Inventory.Packages {
+		pkgPurl := converter.ToPURL(pkg)
+		packageSet[pkgPurl.String()] = pkg
+	}
+
+	for _, pkg := range packageSet {
+		if _, ok := manifests[pkg.Ecosystem()]; !ok {
+			manifests[pkg.Ecosystem()] = models.NewPackageManifestFromLocal("", pkg.Ecosystem())
```

Comment:
We should use PURL here using `NewPackageManifestFromPurl`. The PURL should be referencing the container image being scanned.